node/bft/src/gateway.rs

```diff
@@ -1535,6 +1549,11 @@ impl<N: Network> Gateway<N> {
             warn!("{CONTEXT} Dropping '{peer_addr}' on version {version} (outdated)");
             return Some(DisconnectReason::OutdatedClientVersion);
         }
+        // If the node is in trusted peers only mode, ensure the peer is trusted.
+        if self.trusted_peers_only && !self.is_trusted(peer_addr) {
```

Comment:
`peer_addr` is not necessarily the address we're recognizing the peer with - that one is constructed based on the listener port from the `ChallengeRequest`, and would need to be built here as well; the same applies to the check for bootstrap peers below, where it had been missed

Reply:
I'm still seeing the issue on that commit